### ISSUE TITLE
False positive write_with_newline on raw string

### ISSUE THREAD

**Issue by uniphil** [rust-lang/rust-clippy#3778]
```
$ cargo clippy -V
clippy 0.0.212 (b2601beb 2018-11-27)
```

The `\n` is literal, not a newline character here, since it's a `r"raw"` string:

```
warning: using `write!()` with a format string that ends in a single newline, consider using `writeln!()` instead
   --> src/ast.rs:150:25
    |
150 |                 '\n' => write!(f, r"\n")?,
    |                         ^^^^^^^^^^^^^^^^
    |
    = note: #[warn(clippy::write_with_newline)] on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#write_with_newline
```

**Comment by uniphil** [comment]
fixed in #3781
